client/consul_template.go

```diff
@@ -394,7 +394,7 @@ func (tm *TaskTemplateManager) handleTemplateRerenders(allRenderedTime time.Time
 					tm.config.Hooks.Kill(consulTemplateSourceName, err.Error(), true)
 					return
 				}
-				tm.config.EnvBuilder.SetTemplateEnv(envMap)
+				tm.config.EnvBuilder.MergeTemplateEnv(envMap)
```

Comment:
This is only additive so env vars will never be removed.

I'm unclear why we don't reparse all env templates on any change and simply set the full map each time? That was my intended behavior, and from looking at the code I'm not sure why it doesn't work.

Reply:
Ah good point. At first I wanted to avoid reparsing everything but you are right, certain env's may be conditional. Will fix